src/PositionContext.ts

```diff
@@ -64,38 +65,35 @@ export class PositionContext {
+    private _jsonToken: CachedValue<Json.Token | undefined> = new CachedValue<Json.Token>();
+    private _jsonValue: CachedValue<Json.Value | undefined> = new CachedValue<Json.Value | undefined>();
+    private _tleInfo: CachedValue<TleInfo | undefined> = new CachedValue<TleInfo | undefined>();
+
+    private constructor(deploymentTemplate: DeploymentTemplate) {
+        this._deploymentTemplate = deploymentTemplate;
+    }
 
     public static fromDocumentLineAndColumnIndexes(deploymentTemplate: DeploymentTemplate, documentLineIndex: number, documentColumnIndex: number): PositionContext {
-        assert(deploymentTemplate !== null, "deploymentTemplate cannot be null");
-        assert(deploymentTemplate !== undefined, "deploymentTemplate cannot be undefined");
-        assert(documentLineIndex !== null, "documentLineIndex cannot be null");
-        assert(documentLineIndex !== undefined, "documentLineIndex cannot be undefined");
+        assert(deploymentTemplate);
+        assert(!isNullOrUndefined(documentLineIndex), "documentLineIndex cannot be null/undefined");
         assert(documentLineIndex >= 0, "documentLineIndex cannot be negative");
         assert(documentLineIndex < deploymentTemplate.lineCount, `documentLineIndex (${documentLineIndex}) cannot be greater than or equal to the deployment template's line count (${deploymentTemplate.lineCount})`);
-        assert(documentColumnIndex !== null, "documentColumnIndex cannot be null");
-        assert(documentColumnIndex !== undefined, "documentColumnIndex cannot be undefined");
+        assert(!isNullOrUndefined(documentColumnIndex), "documentColumnIndex cannot be null/undefined");
         assert(documentColumnIndex >= 0, "documentColumnIndex cannot be negative");
         assert(documentColumnIndex <= deploymentTemplate.getMaxColumnIndex(documentLineIndex), `documentColumnIndex (${documentColumnIndex}) cannot be greater than the line's maximum index (${deploymentTemplate.getMaxColumnIndex(documentLineIndex)})`);
 
-        let context = new PositionContext();
-        context._deploymentTemplate = deploymentTemplate;
+        let context = new PositionContext(deploymentTemplate);
         context._givenDocumentPosition = new language.Position(documentLineIndex, documentColumnIndex);
         return context;
 
     }
     public static fromDocumentCharacterIndex(deploymentTemplate: DeploymentTemplate, documentCharacterIndex: number): PositionContext {
-        assert(deploymentTemplate !== null, "deploymentTemplate cannot be null");
-        assert(deploymentTemplate !== undefined, "deploymentTemplate cannot be undefined");
-        assert(documentCharacterIndex !== null, "documentCharacterIndex cannot be null");
-        assert(documentCharacterIndex !== undefined, "documentCharacterIndex cannot be undefined");
+        assert(!isNullOrUndefined(deploymentTemplate), "deploymentTemplate cannot be null/undefined");
```

Comment:
>isNullOrUndefined [](start = 16, length = 17)

Why are we using !isNullOrUndefined on deploymentTemplate here but not in fromDocumentLineAndColumnIndexes above?

Reply:
Fixed